bug: for_each fails with same-module array constants

## Description

Using `for_each` to iterate over an array constant from another file in the same module fails, even when using the module prefix.

## QA Test Results

```
✅ return ITEMS[0]              - WORKS (array index access)
✅ len(ITEMS)                    - WORKS (array length)
✅ range(0, len(ITEMS))          - WORKS (in regular for)
❌ for_each item in ITEMS        - FAILS (no prefix)
❌ for_each item in mymod.ITEMS  - FAILS (with prefix)
```

## Example that FAILS

In `types.ez`:
```ez
module mymod
const ITEMS [string, 3] = {"apple", "banana", "cherry"}
```

In `helpers.ez` (same `mymod` module):
```ez
module mymod

import @std
using std

do print_all_items() {
    for_each item in ITEMS {      // FAILS
        println(item)
    }
}

do print_all_items_v2() {
    for_each item in mymod.ITEMS { // ALSO FAILS
        println(item)
    }
}
```

## Error Messages

Without prefix:
```
error[E2002]: expected :, got ( instead
  --> helpers.ez:10:16
10 |         println(item)
              ^ expected token not found
```

With prefix:
```
error[E4001]: identifier not found: 'mymod'
```

## Expected Behavior

`for_each` should work with same-module array constants, with or without the module prefix.

## Actual Behavior

Parser fails when encountering same-module array constant in `for_each` expression. The error messages point to wrong locations.

## Related

This is related to #462 (same-module constant access issues) but appears to be a separate parsing issue specific to the `for_each` construct.

--- a/integration-tests/pass/multi-file/foreach-module-constant/main.ez
+++ b/integration-tests/pass/multi-file/foreach-module-constant/main.ez
@@ -1,0 +1,40 @@
+/*
+ * Test for_each with same-module array constants
+ * Regression test for bug #465
+ */
+import @std
+import "./testmod"
+
+using std
+
+do main() {
+    println("=== For_each Module Constant Test ===")
+    temp passed int = 0
+    temp failed int = 0
+
+    // Test 1: for_each over string array constant
+    println("Items:")
+    testmod.print_items()
+    println("  [PASS] for_each over string array")
+    passed += 1
+
+    // Test 2: for_each over int array constant with accumulation
+    temp sum = testmod.sum_numbers()
+    if sum == 100 {
+        println("  [PASS] for_each over int array (sum=${sum})")
+        passed += 1
+    } otherwise {
+        println("  [FAIL] for_each over int array: expected 100, got ${sum}")
+        failed += 1
+    }
+
+    // Summary
+    println("")
+    println("Results: ${passed} passed, ${failed} failed")
+
+    if failed > 0 {
+        println("SOME TESTS FAILED")
+    } otherwise {
+        println("ALL TESTS PASSED")
+    }
+}

--- a/integration-tests/pass/multi-file/foreach-module-constant/testmod/constants.ez
+++ b/integration-tests/pass/multi-file/foreach-module-constant/testmod/constants.ez
@@ -1,0 +1,4 @@
+module testmod
+
+const ITEMS [string, 3] = {"apple", "banana", "cherry"}
+const NUMBERS [int, 4] = {10, 20, 30, 40}

--- a/integration-tests/pass/multi-file/foreach-module-constant/testmod/helpers.ez
+++ b/integration-tests/pass/multi-file/foreach-module-constant/testmod/helpers.ez
@@ -1,0 +1,18 @@
+module testmod
+
+import @std
+using std
+
+do print_items() {
+    for_each item in ITEMS {
+        println(item)
+    }
+}
+
+do sum_numbers() -> int {
+    temp total int = 0
+    for_each n in NUMBERS {
+        total += n
+    }
+    return total
+}

